### PR TITLE
A more thorough Hello World restart test.

### PR DESCRIPTION
**Pull request by mcovarr**
Necessitated some actual fixes to the code to do the Right Thing for the LocalBackend.

### CODE REVIEW

File: src/main/scala/cromwell/engine/backend/local/LocalBackend.scala [hunk 1]
```diff
@@ -176,4 +180,27 @@ class LocalBackend extends Backend with LazyLogging {
...
+
+    val seqOfFutures = restartableWorkflows map { workflow =>
+      for {
+        callsToStatuses <- dataAccess.getExecutionStatuses(workflow.id)
+        _ <- updateNonTerminalCalls(workflow.id, callsToStatuses)
+      } yield ()
+    }
+    // The caller doesn't care about the result value, only that it's a Future.
+    // Use Future.sequence to manipulate Seq[Future[Unit]] to Future[Seq[Unit]], then map to Future[Unit].
+    Future.sequence(seqOfFutures) map { _ => () }
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/backend/local/LocalBackend.scala]: Why do you need that map?

Comment (mcovarr) [inline src/main/scala/cromwell/engine/backend/local/LocalBackend.scala]: The expression doesn't typecheck to the return type without it.

Comment (mcovarr) [inline src/main/scala/cromwell/engine/backend/local/LocalBackend.scala]: It would typecheck to `Future[Any]`; maybe I should switch to that?

**Comment by kshakir** [comment]
Looks good. Merging.

**Pull request merged**